Pipeline task 2.51.1 fails with Error: Parameter 'toolPath' cannot be null or empty.

**Describe the bug**
Running Azure DevOps Dependabot task v2.51.1 results in an unhandled exception:
```
Processing job from '/tmp/dependabot-jobs/discover-0-nuget-dependency-list/job.yaml'
##[debug]task result: Failed
##[error]Parameter 'toolPath' cannot be null or empty.
##[debug]Processed: ##vso[task.issue type=error;source=TaskInternal;]Parameter 'toolPath' cannot be null or empty.
##[debug]Processed: ##vso[task.complete result=Failed;]Parameter 'toolPath' cannot be null or empty.
##[error]An unhandled exception occurred: Error: Parameter 'toolPath' cannot be null or empty.
##[debug]Processed: ##vso[task.issue type=error;source=TaskInternal;]An unhandled exception occurred: Error: Parameter 'toolPath' cannot be null or empty.
Error: Parameter 'toolPath' cannot be null or empty.
    at new ToolRunner2 (/home/vsts/work/_tasks/dependabot_d98b873d-cf18-41eb-8ff5-234f14697896/2.51.1/dist/task-v2.js:4077:19)
    at tool2 (/home/vsts/work/_tasks/dependabot_d98b873d-cf18-41eb-8ff5-234f14697896/2.51.1/dist/task-v2.js:6221:16)
    at DependabotCli.update (/home/vsts/work/_tasks/dependabot_d98b873d-cf18-41eb-8ff5-234f14697896/2.51.1/dist/task-v2.js:105280:54)
    at async performDependabotUpdatesAsync (/home/vsts/work/_tasks/dependabot_d98b873d-cf18-41eb-8ff5-234f14697896/2.51.1/dist/task-v2.js:110012:47)
    at async run (/home/vsts/work/_tasks/dependabot_d98b873d-cf18-41eb-8ff5-234f14697896/2.51.1/dist/task-v2.js:109941:24)
Finishing: dependabot
```

**Categorization**
- [x] This is not a permissions issue (Seek help at https://github.com/tinglesoftware/dependabot-azure-devops/discussions/1245)
- [x] This is an issue specific to Azure DevOps or this repository and does not belong in [dependabot-core](https://github.com/dependabot/dependabot-core).
  > Specific issues for dependabot are solved faster in the core repository. For example, why a package version is skipped. Trying out this behaviour in the GitHub Hosted version can help you pinpoint where it lies.
- [ ] I have linked a public reproduction of the specific issue or none is required because the issue is not specific to me.
  > Please note that you can create a public organization/project and repository to show the issue. This tends to accelerate resolution.

**Expected behavior**
Azure DevOps Dependabot task executes without exception.

**Extension (please complete the following information):**
 - Host: Azure DevOps
 - Agent version: 4.255.0
 - Task Version 2.51.1

**Additional context**
This has a last successful run on task version 2.50.49

--- a/.changeset/many-cycles-carry.md
+++ b/.changeset/many-cycles-carry.md
@@ -1,0 +1,5 @@
+---
+'extension-azure-devops': patch
+---
+
+Use ||= instead of ??= when finding go/dependabot tool

--- a/extensions/azure/src/dependabot/cli.ts
+++ b/extensions/azure/src/dependabot/cli.ts
@@ -191,7 +191,7 @@ export class DependabotCli {
   // Get the dependabot tool path and install if missing
   private async getDependabotToolPath(installIfMissing: boolean = true): Promise<string> {
     debug('Checking for `dependabot` install...');
-    this.toolPath ??= which('dependabot', false);
+    this.toolPath ||= which('dependabot', false);
     if (this.toolPath) {
       return this.toolPath;
     }
@@ -209,7 +209,7 @@ export class DependabotCli {
     // If dependabot still cannot be found using `which()` after install, we must manually resolve the path;
     // It will either be "$GOPATH/bin/dependabot" or "$HOME/go/bin/dependabot", if GOPATH is not set.
     const goBinPath = process.env.GOPATH ? path.join(process.env.GOPATH, 'bin') : path.join(os.homedir(), 'go', 'bin');
-    return (this.toolPath ??= which('dependabot', false) || path.join(goBinPath, 'dependabot'));
+    return (this.toolPath ||= which('dependabot', false) || path.join(goBinPath, 'dependabot'));
   }
 
   // Create the jobs directory if it does not exist

